File: app/src/main/java/com/ixidev/simplenotepad/callbacks/MainActionModeCallback.java
package com.ixidev.simplenotepad.callbacks;

import android.view.ActionMode;
import android.view.Menu;
import android.view.MenuItem;

import com.ixidev.simplenotepad.R;

/**
 * Created by ixi.Dv on 22/07/2018.
 */
public abstract class MainActionModeCallback implements ActionMode.Callback {
    private ActionMode action;
    private MenuItem countItem;
    private MenuItem shareItem;

    @Override
    public boolean onCreateActionMode(ActionMode actionMode, Menu menu) {
        actionMode.getMenuInflater().inflate(R.menu.main_action_mode, menu);
        this.action = actionMode;
        this.countItem = menu.findItem(R.id.action_checked_count);
        this.shareItem = menu.findItem(R.id.action_share_note);
        return true;
    }

    @Override
    public boolean onPrepareActionMode(ActionMode actionMode, Menu menu) {
        return false;
    }

    @Override
    public void onDestroyActionMode(ActionMode actionMode) {

    }

    public void setCount(String chackedCount) {
        if (countItem != null)
            this.countItem.setTitle(chackedCount);
    }

    /**
     * if checked count > 1 hide shareItem else show it
     *
     * @param b :visible
     */
    public void changeShareItemVisible(boolean b) {
        shareItem.setVisible(b);
    }

    public ActionMode getAction() {
        return action;
    }
}

File: Kotlin/Application/app/src/main/java/com/ixidev/simplenotepad/callbacks/MainActionModeCallback.kt
package com.ixidev.simplenotepad.callbacks

import android.view.Menu
import android.view.MenuItem
import androidx.appcompat.view.ActionMode
import com.ixidev.simplenotepad.R

/**
 * Created by ixi.Dv on 22/07/2018.
 */
public abstract class MainActionModeCallback : ActionMode.Callback {
    var action: ActionMode? = null
        private set
    private var countItem: MenuItem? = null
    private var shareItem: MenuItem? = null
    override fun onCreateActionMode(
        mode: ActionMode?,
        menu: Menu?
    ): Boolean {
        mode?.let {
            it.menuInflater.inflate(R.menu.main_action_mode, menu)
            action = it
            countItem = menu!!.findItem(R.id.action_checked_count)
            shareItem = menu.findItem(R.id.action_share_note)
        }
        return true
    }

    override fun onPrepareActionMode(
        actionMode: ActionMode,
        menu: Menu
    ): Boolean {
        return false
    }

    override fun onDestroyActionMode(actionMode: ActionMode) {}


    fun setCount(chackedCount: String?) {
        countItem?.apply {
            title = chackedCount
        }
    }

    /**
     * if checked count > 1 hide shareItem else show it
     *
     * @param b :visible
     */
    fun changeShareItemVisible(b: Boolean) {
        shareItem?.apply {
            isVisible = b
        }
    }

}